vscode-wpilib/src/dependencyView.ts

```diff
@@ -0,0 +1,435 @@
+import * as vscode from 'vscode';
+import * as fetch from 'node-fetch';
+import { ProjectInfoGatherer, IProjectInfo } from './projectinfo';
+import { VendorLibraries } from './vendorlibraries';
+import { IJsonDependency } from './shared/vendorlibrariesbase';
+import { IExternalAPI } from 'vscode-wpilibapi';
+import { isNewerVersion } from './versions';
+import { logger } from './logger';
+import { localize as i18n } from './locale';
+
+export interface IJsonList {
+  path: string;
+  name: string;
+  version: string;
+  uuid: string;
+  description: string;
+  website: string;
+}
+
+export interface IDepInstalled { name: string; currentVersion: string; versionInfo: { version: string, buttonText: string }[]; }
+
+export interface IJSMessage { type: string; version: string; index: string; }
+
+export class DependencyViewProvider implements vscode.WebviewViewProvider {
+  public static readonly viewType = 'wpilib.dependencyView';
+  private projectInfo: ProjectInfoGatherer;
+  private vendorLibraries: VendorLibraries;
+  private viewInfo: IProjectInfo | undefined;
+  private disposables: vscode.Disposable[] = [];
+  private installedDeps: IJsonDependency[] = [];
+  private availableDeps: IJsonList[] = [];
+  private availableDepsList: IJsonList[] = [];
+  private onlineDeps: IJsonList[] = [];
+  private installedList: IDepInstalled[] = [];
+  private homeDeps: IJsonDependency[] = [];
+  private externalApi: IExternalAPI;
+  private ghURL = `https://raw.githubusercontent.com/jasondaming/vendor-json-repo/ctre2024/`;
```

Comment:
This should change to the WPILIb repo before merging

Reply:
Should I do this now even though 2024 / 2025 don't exist currently?